test/unit/constraints/array.test.ts

```diff
@@ -0,0 +1,113 @@
+import "../metadataShim";
+import {registerJoi, WORKING_SCHEMA_KEY} from "../../../src/core";
+import * as Joi from 'joi';
+import { Max, Length, ArraySchema, Min, Unique } from "../../../src/constraints/array";
+import {Validator} from "../../../src/Validator";
+import { testConstraint } from "../testUtil";
+
+
+registerJoi(Joi);
+
+describe('Array Test', () => {
+    describe('Test Array methods ', () => {
+        it('Should annotate class', () => {
+            class MyClass {
+                @ArraySchema()
+                myProperty!: Array<number>
+            }
+            const metadata = Reflect.getMetadata(WORKING_SCHEMA_KEY, MyClass.prototype);
+            const expected = {
+                myProperty: Joi.array()
+            }
+            expect(metadata).toEqual(expected)
+        })
+        it('Should allow correct max value for array', () => {
```

Comment:
:pray: Existing constraint tests try to use the `testConstraint()` function, where possible. This cuts down on boilerplate, and enforces that both postivie and negative tests are implemented. I see you've already used it for the `@Unique()` tests, let's keep it consistent and use if for `@Max()`, `@Length()`, and `@Min()` as well.